sinks/src/main/scala/com.snowplowanalytics.snowplow.eventgen/Main.scala

```diff
@@ -109,44 +114,78 @@ object Main extends IOApp {
       }
     }
 
-    def sinkFor(name: String, idx: Int, predicate: Boolean): Pipe[F, Byte, INothing] =
+    def sinkFor(name: String, idx: Int, predicate: Boolean): Pipe[F, Byte, Nothing] =
       if (predicate) sinkBuilder(name, idx)
       else _.drain
 
+    def fileSink: Pipe[F, GenOutput, Unit] =
+      RotatingSink.rotate(config.payloadsPerFile) { idx =>
+        val pipe1: Pipe[F, GenOutput, Nothing] =
+          _.map(_._1).through(Serializers.rawSerializer(config.compress)).through(sinkFor("raw", idx, config.withRaw))
+        val pipe2: Pipe[F, GenOutput, Nothing] = _.flatMap(in => Stream.emits(in._2))
+          .through(Serializers.enrichedTsvSerializer(config.compress))
+          .through(sinkFor("enriched", idx, config.withEnrichedTsv))
+        val pipe3: Pipe[F, GenOutput, Nothing] = _.flatMap(in => Stream.emits(in._2))
+          .through(Serializers.enrichedJsonSerializer(config.compress))
+          .through(sinkFor("transformed", idx, config.withEnrichedJson))
+        in: Stream[F, GenOutput] => in.broadcastThrough(pipe1, pipe2, pipe3)
```

Comment:
It seems to me that with this way of doing, we do serialize in raw, TSV and JSON even though one of these is not activated. Am I wrong @istreeter ?

Reply:
Oh yeah, well spotted.  That could be improved.